tasks/aiida-prepare.yml

```diff
@@ -1,19 +1,47 @@
 ---
-- name: Make sure Postgresql service is started
+- name: Check if Postgresql is running
+  shell: pgrep postgres
+  changed_when: false
+  register: postgres_status
+  failed_when: postgres_status.rc not in [0,1]
+  when: ansible_service_mgr != "systemd"
+
+- name: Start Postgresql service
   become: true
   become_user: "{{ root_user }}"
   service:
     name: "postgresql"
     state: "started"
     enabled: true
+  when: ansible_service_mgr == "systemd"
 
-- name: Make sure RabbitMQ service is started
+- name: Start Postgresql
+  become: true
+  become_user: "{{ aiida_postgres_user }}"
+  command: "{{ aiida_postgres_bin_location }}/pg_ctl start -D {{ aiida_postgres_data_location }} -s -o '-p 5432' -w -t 300"
+  when: postgres_status.rc !=0 and ansible_service_mgr != "systemd"
+
+- name: Check if RabbitMQ is running
+  shell: pgrep rabbitmq
+  changed_when: false
+  register: rabbitmq_status
+  failed_when: rabbitmq_status.rc not in [0,1]
+  when: ansible_service_mgr != "systemd"
```

Comment:
perhaps move the non-systemd stuff into a 
```yaml
- block:
   - name: Check if ...
   - name: Start p
   when: ansible_service_mgr != "systemd"
```

Reply:
That is certainly a good suggestion. Done.